### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by squeedee**
Just 2 quick houskeeping steps

1. Introduce PHONIES to avoid future confusion with our Makefiles
2. Lint the hack scripts (to the level of error) with shellcheck.

### CODE REVIEW

File: hack/setup.sh [hunk 1]
```diff
@@ -114,7 +114,7 @@ show_usage_help() {
 display_vars() {
         echo "Variables:
 
-	COMMANDS: 	$@
+	COMMANDS: 	$*
```
Comment (cirocosta) [inline hack/setup.sh:117]: is this from shellsheck? IIUC, $@ and $* don't really matter in this particular case

File: hack/Makefile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.PHONY: lint
+
+lint:
+	shellcheck *.sh -S error # eventually we want to lift the severity and catch warnings too.
```
Comment (cirocosta) [inline hack/Makefile:4]: I think we should update `./CONTRIBUTING.md` - there we mention all the dev dependencies needed

File: Makefile [hunk 3]
```diff
@@ -1,12 +1,18 @@
+.PHONY: build
```
Comment (cirocosta) [inline Makefile:1]: I think `phony'`ing `build `is indeed necessary, but, what'd be the reason for the others? I'm not suuuper familiar with `make` details like this, but from what I understand, as the other targets don't produce anything that matches a target, we'd not have nay executions being skipped

**Review by cirocosta — APPROVED** [review]
lgtm  👍  just left a few questions

File: hack/Makefile (hunk 2 above)
Comment (cirocosta) [reply hack/Makefile:4]: hmmm I'm surprised this didn't break `ci` 🤔  https://github.com/vmware-tanzu/cartographer/blob/cce05c1d5853deef994635a62eb4d2b11d7fbac1/.github/workflows/validation.yaml#L46

I guess that comes installed  by default on their ubuntu setup?

https://github.com/vmware-tanzu/cartographer/runs/3723206626#step:5:247

File: hack/setup.sh (hunk 1 above)
Comment (squeedee) [reply hack/setup.sh:117]: Yeah it was from shellcheck

File: hack/Makefile (hunk 2 above)
Comment (squeedee) [reply hack/Makefile:4]: That's a good point. I guess it must?

File: Makefile (hunk 3 above)
Comment (squeedee) [reply Makefile:1]: The reason is future proofing. The day you create a file that matches a Make target is the day you spend hours trying to work out why something isn't triggering.

Comment (squeedee) [reply Makefile:1]: By way of example, you accidentally create a file called test. This is common as in `touch foo` or `vim test`

Then you finish your work, add the files you care about `make test && git push`

Comment (cirocosta) [reply Makefile:1]: got it, makes sense

thanks!

**Review by cirocosta — APPROVED** [review]
lgtm 😁

**Pull request merged**